### PR TITLE
Support for request logging in kafka emitter.

### PR DESCRIPTION
**Pull request by mkuthan**
This is continuation of #5782. Due to Apache Incubating process and packages renames it was easier to create new PR than resolve conflicts, sorry for inconvenience.

@jihoonson According to your comment the Jackson mixin annotation has been moved into kafka-emitter module.

### CODE REVIEW

**Comment by gianm** [comment]
By the way, apologies for the delay in picking review back up on this PR>

**Comment by gianm** [comment]
Hey @mkuthan  - hope the review was helpful, let us know if you have any questions about further implementation!

File: extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java [hunk 1]
```diff
@@ -47,13 +49,15 @@ public KafkaEmitterConfig(
...
       @JsonProperty("alert.topic") String alertTopic,
+      @JsonProperty("request.topic") String requestTopic,
       @JsonProperty("clusterName") String clusterName,
       @JsonProperty("producer.config") @Nullable Map<String, String> kafkaProducerConfig
   )
   {
     this.bootstrapServers = Preconditions.checkNotNull(bootstrapServers, "bootstrap.servers can not be null");
     this.metricTopic = Preconditions.checkNotNull(metricTopic, "metric.topic can not be null");
     this.alertTopic = Preconditions.checkNotNull(alertTopic, "alert.topic can not be null");
+    this.requestTopic = Preconditions.checkNotNull(requestTopic, "request.topic can not be null");
```
[the remark this answers is not in the document]
Comment (mkuthan) [reply extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterConfig.java]: We should be consistent with all 3 topics and make all of them required or optional, e.g. it is fully valid scenario when we are interested in metrics but not in alerts. 

I agree that should be a waring if emitter receives event but corresponding topic is not configured.

File: extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterModule.java [hunk 2]
```diff
@@ -56,3 +62,12 @@ public Emitter getEmitter(KafkaEmitterConfig kafkaEmitterConfig, ObjectMapper ma
     return new KafkaEmitter(kafkaEmitterConfig, mapper);
   }
 }
+
+@JsonAppend(
+    attrs = {
+        @JsonAppend.Attr(value = "clusterName")
+    }
+)
+abstract class ClusterNameMixin
```
[the remark this answers is not in the document]
Comment (mkuthan) [reply extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitterModule.java]: DefaultRequestLogEvent.toMap returns empty map for good reason.
https://github.com/apache/incubator-druid/blob/2cdce2e2a6111acdd8fffd4ef61edc1c89f187bf/server/src/main/java/org/apache/druid/server/log/DefaultRequestLogEvent.java#L59

**Comment by Raboo** [comment]
This is sad that this pr never was merged. It would be awesome if someone would pick this up. I can contribute with lots of emojis if that happens!

**Comment by himanshug** [comment]
this looks like very close indeed, @mkuthan there is at least  one more person ( @Raboo ) needing this , if you wanna revive this PR  , I can help review and get in.

**Pull request closed without merging**